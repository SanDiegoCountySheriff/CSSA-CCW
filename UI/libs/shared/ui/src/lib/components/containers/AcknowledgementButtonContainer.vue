<template>
  <v-card-actions>
    <v-btn
      @click="onDecline"
      color="error"
      text
      absolute
      bottom
    >
      {{ $t('Decline') }}
    </v-btn>
    <v-btn
      :disabled="isLoading"
      @click="onAccept"
      color="primary"
      text
      absolute
      bottom
      right
    >
      {{ $t('Accept') }}
    </v-btn>
  </v-card-actions>
</template>

<script setup lang="ts">
interface IAcknowledgmentButtonContainerProps {
  isLoading?: boolean
}
const props = withDefaults(defineProps<IAcknowledgmentButtonContainerProps>(), {
  isLoading: false,
})

const emit = defineEmits(['handle-accept', 'handle-decline'])

function onAccept() {
  emit('handle-accept')
}

function onDecline() {
  emit('handle-decline')
}
</script>
