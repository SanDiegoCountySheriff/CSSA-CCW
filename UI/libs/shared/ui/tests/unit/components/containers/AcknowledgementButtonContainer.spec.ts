/* eslint-disable @typescript-eslint/ban-ts-comment */
import AcknowledgementButtonContainer from '@shared-ui/components/containers/AcknowledgementButtonContainer.vue';
import Vuetify from 'vuetify';
import { createLocalVue, mount } from '@vue/test-utils';

const localVue = createLocalVue();
const tMock = {
  $t: text => text,
};

describe('AcknowledgementButtonContainer', () => {
  let vuetify;
  let wrapper;

  beforeAll(() => {
    vuetify = new Vuetify();
    //@ts-ignore
    wrapper = mount(AcknowledgementButtonContainer, {
      localVue,
      vuetify,
      mocks: tMock,
    });
  });

  it('should match the snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
