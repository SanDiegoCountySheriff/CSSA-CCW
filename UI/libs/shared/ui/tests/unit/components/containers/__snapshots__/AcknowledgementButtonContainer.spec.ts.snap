// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AcknowledgementButtonContainer should match the snapshot 1`] = `
<div class="acknowledgement-button-container"><button type="button" class="v-btn v-btn--is-elevated v-btn--has-bg theme--light v-size--default success mr-3"><span class="v-btn__content">
    Accept
  </span></button> <button type="button" class="v-btn v-btn--is-elevated v-btn--has-bg theme--light v-size--default error mr-3"><span class="v-btn__content">
    Decline
  </span></button></div>
`;
